src/spikeinterface/core/basesorting.py

```diff
@@ -496,6 +496,14 @@ def precompute_spike_trains(self, from_spike_vector=None):
                 for unit_id in unit_ids:
                     self.get_unit_spike_train(unit_id, segment_index=segment_index, use_cache=True)
 
+    def _custom_cache_spike_vector(self) -> None:
+        """
+        Function that can be implemented by some children sorting to quickly
+        cache the spike vector without computing it from spike trains
+        (e.g. computing it from a sorting parent).
+        """
+        pass
```

Comment:
One tiny question: 

If the child class needs to implement why a `pass` and not a `raise NotImplementedError`?

Reply:
The child class does not need to implement it :)

Reply:
So you're relying on the `_` to say it is a private function that people will never accidently call and have nothing happen? 

To better explain my point:
 If someone accidently calls a function I would expect it to say this child class doesn't not support use of this function which is indicated by a `NotImplementedError`. But if this is private then I suppose you are saying that if someone uses they are doing so at their own risk of this failing?

And as a second point what if I as a developer want to add a "failing" test for this so I can implement my own function to have the test pass (TDD), then if it has built in pass I can't use this strategy.

Again, I'm just curious about the strategy and I really appreciate the docstring!

Reply:
Okay cool. I don't work at this level with SI that much (and I know you do this stuff all the time) so thanks for the explanation :)